### ISSUE TITLE
Could have won at 3 player chinese checkers but didn't!

### ISSUE THREAD

**Issue by SteveDraper** [SanchoGGP/ggp-base#253]
On turn 28 of this match: http://www.ggp.org/view/tiltyard/matches/12eecdd60a20307304136c49d3631bb24cf2d8b2/
we played 'noop'.  As far as I can see we had a win (4 moves).  Greenshell can also compleet in 4, but we had the first move.  Why did we not take it!  Player output was claiming all moves lead to the same score (50), but that doesn't look correct to me.

**Comment by arr28** [comment]
More oddly, the score isn't even 50 - it's 48.15 (selection 47.99) for all the moves.  How can a move be marked complete and have a score other than 0/25/50/100?

```
5/19/2015, 1:54:27 AM.729	INFO	Move noop scores 48.15 (selectionScore score 47.99, selection count 19086, ref 223339208866, complete [11])
5/19/2015, 1:54:27 AM.729	INFO	Move ( move g4 h2 ) scores 48.15 (selectionScore score 47.99, selection count 11146, ref 231930774373, complete [11])
5/19/2015, 1:54:27 AM.729	INFO	Move ( move g4 h1 ) scores 48.15 (selectionScore score 47.99, selection count 12350, ref 274880017658, complete [11])
5/19/2015, 1:54:27 AM.729	INFO	Move ( move e3 f4 ) scores 48.15 (selectionScore score 47.99, selection count 74644, ref 223338777652, complete [13])
5/19/2015, 1:54:27 AM.729	INFO	Move ( move e3 f3 ) scores 48.15 (selectionScore score 47.99, selection count 48437, ref 240519071229, complete [13])
5/19/2015, 1:54:27 AM.729	INFO	Move ( move g3 h1 ) scores 48.15 (selectionScore score 47.99, selection count 15396, ref 197571172256, complete [11])
5/19/2015, 1:54:27 AM.729	INFO	1 level minimax result differed from best raw move: org.ggp.base.player.gamer.statemachine.sancho.TreeEdge@61dea5d2
5/19/2015, 1:54:27 AM.729	INFO	Most likely path: noop, ( move c2 c1 ) | ( move d4 d5 ) | ( move e3 f3 ), ( move e4 e3 ) | noop | ( move f3 h2 ), noop | noop | ( move g4 h1 ), noop | noop | ( move g3 i1 ),
```

**Comment by SteveDraper** [comment]
That part is fine.  Don;t forget we add a small win/loss bias to the actual
score so as to prefer 50 with a higher relative placement to opponents over
50 with a lower one.

**Comment by SteveDraper** [comment]
Turns out that in fact all branches do have equal scores, but only if the two, opponents collude to engineer a win for one of them, with no net-benefit for the other.  Node completion was assuming pessimal behavior for us in the event of a tie for the chooser, but that's actually an overly pessimistic viewpoint, and results in us just giving up and playing noops (or other meaningless moves) like this.  I have modified (somewhat experimentally) the code to instead complete with a blended score of the equal choices for the chooser, weighted by a weight inversely correlated with OUR score - so still pessimistic, but also blending in an element of the other chooser-equal choices which are not pessimal for us.  This allows us to differentiate can-win-in-some-paths-if-opponents-don't-cooperate moves from can-never-win-if-opponents-try-for-own-wins paths, and play a little more optimistically.
